### ISSUE TITLE
SystemStackError crash in conversion substitution

### ISSUE THREAD

**Issue by zelivans** [asciidoctor/asciidoctor#2883]
Second bug in `substitutors.rb` found with the fuzzer.

## Raw crash
```
/home/ariel/afl-kisaten/private/sandbox/asciidoctor/asciidoctor/lib/asciidoctor/substitutors.rb:106:in `each': stack level too deep (SystemStackError)
	from /home/ariel/afl-kisaten/private/sandbox/asciidoctor/asciidoctor/lib/asciidoctor/substitutors.rb:106:in `apply_subs'
	from /home/ariel/afl-kisaten/private/sandbox/asciidoctor/asciidoctor/lib/asciidoctor/substitutors.rb:350:in `block in restore_passthroughs'
	from /home/ariel/afl-kisaten/private/sandbox/asciidoctor/asciidoctor/lib/asciidoctor/substitutors.rb:347:in `gsub'
	from /home/ariel/afl-kisaten/private/sandbox/asciidoctor/asciidoctor/lib/asciidoctor/substitutors.rb:347:in `restore_passthroughs'
	from /home/ariel/afl-kisaten/private/sandbox/asciidoctor/asciidoctor/lib/asciidoctor/substitutors.rb:354:in `block in restore_passthroughs'
	from /home/ariel/afl-kisaten/private/sandbox/asciidoctor/asciidoctor/lib/asciidoctor/substitutors.rb:347:in `gsub'
	from /home/ariel/afl-kisaten/private/sandbox/asciidoctor/asciidoctor/lib/asciidoctor/substitutors.rb:347:in `restore_passthroughs'
	from /home/ariel/afl-kisaten/private/sandbox/asciidoctor/asciidoctor/lib/asciidoctor/substitutors.rb:354:in `block in restore_passthroughs'
	 ... 10067 levels...
	from /home/ariel/afl-kisaten/private/sandbox/asciidoctor/asciidoctor/lib/asciidoctor/converter/html5.rb:278:in `embedded'
	from /home/ariel/afl-kisaten/private/sandbox/asciidoctor/asciidoctor/lib/asciidoctor/converter/base.rb:38:in `convert'
	from /home/ariel/afl-kisaten/private/sandbox/asciidoctor/asciidoctor/lib/asciidoctor/document.rb:1162:in `convert'
	from tst.rb:6:in `<main>'
```

## Explanation
The input is similar to https://github.com/asciidoctor/asciidoctor/issues/2882, but crashes asciidoctor with stack level too deep (`SystemStackError`). I did not investigate this too much so I have no fix proposed for this bug.

Reproduce with:
```
[']\+++++++++This++++++++++++
```

**Comment by mojavelinux** [comment]
I'm considering this a duplicate of #2882, though I'll incorporate both cases into the test suite.

**Comment by zelivans** [comment]
Sounds right, I opened two bugs because I didn't know if the crash was resulting from the same line. Thanks @mojavelinux

**Comment by mojavelinux** [comment]
You did the right thing. We just happened to discover that the same fix corrected both problems. What's even more interesting is that it encountered two additional bugs as well; a) substitutions were being applied to attribute values that started with an apostrophe but did not end with one and b) attributes associated with a passthrough were being parsed differently than attributes associated with other types of formatted (aka quoted) text.

**Comment by mojavelinux** [comment]
This was closed when #2882 was closed.
